### PR TITLE
add restarting workspace in debug mode

### PR DESCRIPTION
**Pull request by olexii4**
Signed-off-by: Oleksii Orel <oorel@redhat.com>

### What does this PR do?
Added restarting workspaces in debug mode after startup failure.

### What issues does this PR fix or reference?
https://github.com/eclipse/che/issues/16050

Screenshot from 2020-03-18 15-12-06
Screenshot from 2020-03-18 15-11-34

You can tested it with  the next che-server image:  olexii4dockerid/che-server:nightly

Screenshot from 2020-03-19 13-19-46

### CODE REVIEW

**Comment by olexii4** [comment]
I tested it with the next devfile:
```
apiVersion: 1.0.0
metadata:
  name: angular-uq3pt
components:
  - mountSources: true
    endpoints:
      - name: angular
        port: 4200
    command:
      - exit
    args:
      - '1'
    memoryLimit: 1Gi
    type: dockerimage
    image: 'quay.io/eclipse/che-nodejs10-community:7.10.0'
    alias: nodejs
```
output:
[debug_logs.txt](https://github.com/eclipse/che-workspace-loader/files/4351755/debug_logs.txt)

File: src/loader/loader.ts [hunk 1]
```diff
@@ -37,7 +36,19 @@ export class Loader {
...
-        document.getElementById('workspace-loader-reload')!.style.display = 'block';
+        const reloadEl = document.getElementById('workspace-loader-reload');
+        if (reloadEl) {
+            const {pathname, search} = window.location;
+            const isDebugMode = search.includes(DEBUG_PARAM);
+            let href: string;
+            if (search === '') {
+                href = `${pathname}?${DEBUG_PARAM}`;
+            } else {
+                href = `${pathname}${search}${!isDebugMode ? `&${DEBUG_PARAM}` : ''}`;
```
Comment (akurinnoy) [inline src/loader/loader.ts]: wondering how this should work... I would avoid using string interpolation inside ternary operator inside string interpolation.

Comment (olexii4) [reply src/loader/loader.ts]: It works perfectly. I tested. But I agree to simplify.
```
                if (isDebugMode) {
                    href = `${pathname}${search}`;
                } else {
                    href = `${pathname}${search}&${DEBUG_PARAM}`;
                }
```
Is it OK solution for you?

Comment (olexii4) [reply src/loader/loader.ts]: Done

**Pull request merged**